LayoutTestBase/Core/LYTViewProvider.h

```diff
@@ -33,7 +33,7 @@ NS_SWIFT_NAME(ViewProvider)
  \returns Template for the data to test on.
  */
 @required
-+ (NSDictionary *)dataSpecForTest;
++ (nullable NSDictionary *)dataSpecForTestWithError:(NSError * _Nullable *)error;
```

Comment:
This is probably obvious (and surely related to making this throwable in Swift), but why are we passing error into the data spec now? 

I'm pretty strongly against making the return types for `dataSpec` and `viewForData` nullable. It seems contradictory to mark these as required and then allow for nil, otherwise there would be nothing to test?

Also the doc comment here needs updating with the new parameters for `viewForData`. Similar at:
*  https://linkedin.github.io/LayoutTest-iOS/pages/020_writingTest.html
* https://linkedin.github.io/LayoutTest-iOS/pages/090_testingOtherObjects.html

as well as in the README

Reply:
The way Swift / Objc interoperability work is by marking the return value as `nullable` and pass `NSError **` as a param to the method. This method signature in Swift is as follow:

```swift
func viewForData(data: [NSObject: AnyObject],
                    reuseView: UIView?,
                         size: LYTViewSize?,
                      context: AutoreleasingUnsafeMutablePointer<AnyObject?>) throws -> UIView
```

So it should either return a view or throw an exception. Apple uses this pattern and suggested to use this to make the Objc more compatible with Swift, for example:

```objc
- (nullable NSArray<NSString *> *)contentsOfDirectoryAtPath:(NSString *)path error:(NSError **)error
```

Also, when we add the support for this method now creates an error object and assign it to error, it should be able to return `nil` as in that case, there's no return value.

Great catch on the documents, I didn't know we have those, updates the instances of the methods with the new signature

Reply:
Awesome makes sense to me. Thanks for clearing it up